assets/js/base/components/tabs/index.js

```diff
@@ -1,84 +1,82 @@
  */
 import './style.scss';
-import TabButton from './tab-button';
 
 const Tabs = ( {
 	className,
 	onSelect = () => null,
 	tabs,
 	activeClass = 'is-active',
 	initialTabName,
-	instanceId,
 	ariaLabel = __( 'Tabbed Content', 'woo-gutenberg-products-block' ),
 	children,
+	instanceId,
 	id,
 } ) => {
-	const [ selected, setSelected ] = useState(
-		initialTabName || ( tabs.length > 0 ? tabs[ 0 ].name : '' )
-	);
-	if ( ! selected ) {
+	const tabState = useTabState( {
+		selectedId: `${ instanceId }-${
+			initialTabName ? initialTabName : tabs[ 0 ].name
+		}`,
+	} );
+	if ( tabs.length === 0 ) {
 		return null;
 	}
-	const handleClick = ( tabKey ) => {
-		setSelected( tabKey );
-		onSelect( tabKey );
-	};
-	const selectedTab = tabs.find( ( tab ) => tab.name === selected );
-	if ( ! selectedTab ) {
+	if ( ! tabState.selectedId ) {
 		throw new Error( 'There is no available tab for the selected item' );
 	}
-	const selectedId = `${ instanceId }-${ selectedTab.name }`;
 	return (
-		<div
-			className={ classnames( 'wc-block-components-tabs', className ) }
-			id={ id }
-		>
-			<div
-				role="tablist"
+		<div className={ classnames( 'wc-block-components-tabs', className ) }>
+			<TabList
+				{ ...tabState }
+				id={ id }
+				className={ 'wc-block-components-tabs__list' }
 				aria-label={ ariaLabel }
-				className="wc-block-components-tabs__list"
 			>
-				{ tabs.map( ( tab ) => (
-					<TabButton
+				{ tabs.map( ( { name, title, ariaLabel: tabAriaLabel } ) => (
+					<Tab
+						{ ...tabState }
+						id={ `${ instanceId }-${ name }` }
+						manual={ true }
 						className={ classnames(
 							'wc-block-components-tabs__item',
-							tab.className,
 							{
-								[ activeClass ]: tab.name === selected,
+								[ activeClass ]:
+									// reakit uses the ID as the selectedId
+									`${ instanceId }-${ name }` ===
+									tabState.selectedId,
 							}
 						) }
-						tabId={ `${ instanceId }-${ tab.name }` }
-						aria-controls={ `${ instanceId }-${ tab.name }-view` }
-						selected={ tab.name === selected }
-						key={ tab.name }
-						ariaLabel={ tab.ariaLabel || null }
-						onClick={ () => handleClick( tab.name ) }
+						onClick={ () => onSelect( name ) }
+						type="button"
+						key={ name }
+						aria-label={ tabAriaLabel }
 					>
-						{ tab.title }
-					</TabButton>
+						<span className="wc-block-components-tabs__item-content">
+							{ title }
+						</span>
+					</Tab>
 				) ) }
-			</div>
-			{ selectedTab && (
-				<div
-					aria-labelledby={ selectedId }
-					role="tabpanel"
-					id={ `${ selectedId }-view` }
+			</TabList>
+
+			{ tabs.map( ( { name } ) => (
+				<TabPanel
+					{ ...tabState }
+					key={ name }
+					id={ `${ instanceId }-${ name }-view` }
+					tabId={ `${ instanceId }-${ name }` }
 					className="wc-block-components-tabs__content"
-					tabIndex={ 0 }
 				>
-					{ children( selected ) }
-				</div>
-			) }
+					{ children( name ) }
+				</TabPanel>
+			) ) }
```

Comment:
If we're mapping anyways, I think we should just pass the content as an object from the PaymentMethods component and just have our `<Tabs>` component receive the content as a part of the `tabs` object (where panel content would be on the `content` prop.  It'll be less confusing.

The only reason I used a render prop for children originally was for lazy rendering of the tab content on demand, but the current implementation renders all tabs at once (which might be better for screen readers maybe).